ME0 Tracker muons

dnash86 opened the pull request:
This PR includes all the code for the addition of ME0 type global muons. It has been tested with "runTheMatrix.py --job-reports -j 8 -s -l 1306.0,101.0,1003.0,50202.0,9.0,25202.0" in CMSSW_8_0_0_pre6 (with the creation of ME0 type muons off, which is default).  Please check for any changes that need to be made and any additional tests that should be performed.

Thanks a lot.

dnash86 committed:
DataFormats for ME0Id

dnash86 committed:
MuonIdentification package for ME0Id

dnash86 committed:
MuonDetIdAssociator

dnash86 committed:
Changes to MuonIdProducer

dnash86 committed:
Changes for compilation, testing

dnash86 committed:
Working on bad_alloc err

dnash86 committed:
Merged ME0-InFullMuonReco-GlobalMuons from repository dnash86

dnash86 committed:
Working reco step without validation

dnash86 committed:
Changes for customs files

dnash86 committed:
Working on test config

dnash86 committed:
Restore removed components

dnash86 committed:
Fix to me0 geo builder

dnash86 committed:
fillDescriptions added

dnash86 committed:
Changes for compilation

dnash86 committed:
ME0 false for now

dnash86 committed:
revert MuonIdProducer

dnash86 committed:
ME0 enable for MuonDetIdAssociator

dnash86 committed:
ME0 enable in TrackAssociatorESProducer

dnash86 committed:
Remove some old comments

cmsbuild commented:
A new Pull Request was created by  @dnash86  for CMSSW_8_0_X.

It involves the following packages:

DataFormats/MuonReco
RecoLocalMuon/GEMRecHit
RecoMuon/DetLayers
RecoMuon/MuonIdentification
SLHCUpgradeSimulations/Configuration
TrackingTools/TrackAssociator


@civanch, @cvuosalo, @mdhildreth, @cmsbuild, @slava77, @davidlange6 can you please review it and eventually sign? Thanks.
@battibass, @makortel, @abbiendi, @GiacomoSguazzoni, @jhgoh, @VinInn, @Martin-Grunewald, @bellan, @rovere, @amagitte, @cerati, @gpetruc, @istaslis, @trocino, @dgulhan, @bachtis, @rociovilar this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

slava77 commented:
@cmsbuild please test

@dnash86 
This will likely go only to 81X release.

cmsbuild commented:
The tests are being triggered in jenkins.

cvuosalo commented:
@dnash86: Could you please give me complete instructions about how to run a test with ME0 muons fully enabled? 
Thanks.

cvuosalo commented on DataFormats/MuonReco/src/Muon.cc:
```diff
@@ -68,6 +82,12 @@ int Muon::numberOfMatches( ArbitrationType type ) const
+      
+      if(type == ME0HitAndTrackArbitration) {
+         if(chamberMatch->me0Matches.empty()) continue;
+         matches += chamberMatch->me0Matches.size();
```
Line 87 is unnecessary. If the vector is `empty()`, then `size()` will equal zero. Line 87 can be deleted, and, in the case of an empty vector, zero will be added to `matches`, to no effect.

cvuosalo commented on DataFormats/MuonReco/src/Muon.cc:
```diff
@@ -247,6 +299,45 @@ unsigned int Muon::RPClayerMask( ArbitrationType type ) const
+      //chamberMatch->station() on me0s now returns "1" no matter what, since there is only one station
+      //This means as it is written now there is a bit shift equal to which number layer the me0Match is
+      //ME0Segments have detId.layer() of 1 every time, with a roll of "-1"
+      if (region==1 && region==-1) {
```
I'm guessing you meant `rollID == -1` here rather than `region == -1`.

dnash86 replied:
I am not sure if rollID == -1 was meant, this may be a check on the region and the layer.  I need to confirm this and come back to this to fix it.

dnash86 replied:
Nevermind, the typo is actually the "&&", it should be an "||".  It is meant as a *probably* unnecessary check that the region is either 1 (plus endcap) or -1 (minus endcap).  Fixed in upcoming push.

cvuosalo commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -452,6 +457,24 @@ void MuonIdProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+   std::auto_ptr<reco::IsoDepositMap> hoDepMap(new reco::IsoDepositMap());
+   reco::IsoDepositMap::Filler hoDepFiller(*hoDepMap);
+   std::auto_ptr<reco::IsoDepositMap> jetDepMap(new reco::IsoDepositMap());
+   reco::IsoDepositMap::Filler jetDepFiller(*jetDepMap);
```
`auto_ptr` is now deprecated. You should use `unique_ptr` or `shared_ptr` instead.

dnash86 replied:
It looks like I was editing an older version of the muonIdProducer that does not use "fillmap", like the modern one in 80X does.  It was not our intention to alter that functionality, I assume I should revert this section as well as this one: https://github.com/cms-sw/cmssw/pull/13308/files#diff-89dc8f122700a0c3b857022fe730fd42L728

Should I go ahead and revert those?

cvuosalo commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -564,23 +597,38 @@ void MuonIdProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
-           }
-         }
+	   if (doME0_){
+	     if ( goodTrackerMuon || goodRPCMuon || goodME0Muon){
```
Unless I'm missing something, I think the duplicate code lines 621-628 can be eliminated by condensing lines 611 & 612:
`if (goodTrackerMuon || goodRPCMuon || (doMeO_ && goodME0Muon)) {`

cvuosalo commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -799,15 +858,24 @@ void MuonIdProducer::fillMuonId(edm::Event& iEvent, const edm::EventSetup& iSetu
-   if ( ! fillMatching_ && ! aMuon.isTrackerMuon() && ! aMuon.isRPCMuon() ) return;
+
+   if (doME0_){
+     if ( ! fillMatching_ && ! aMuon.isTrackerMuon() && ! aMuon.isRPCMuon() && ! aMuon.isME0Muon() ) return;
```
Lines 865-867 can be deleted by condensing lines 862 & 863:
`if ( ! fillMatching_ && ! aMuon.isTrackerMuon() && ! aMuon.isRPCMuon() && (! doME0_ || ! aMuon.isME0Muon()) )`

cvuosalo commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -925,6 +992,68 @@ void MuonIdProducer::fillMuonId(edm::Event& iEvent, const edm::EventSetup& iSetu
+
+   		 //These matches are hardcoded, is that okay?
+   		 if( (AbsDx <= 3 or AbsDx/sqrt(localError.xx() + me0RecHit.localPositionError().xx()) <= 4) and 
+   		     (AbsDy <= 9999. or AbsDy/sqrt(localError.yy() + me0RecHit.localPositionError().yy()) <= 9999.)
```
Are the local errors here and in line 1047 guaranteed to be > 0, even with corrupted data? If not, please check for > 0 before dividing by them.

dnash86 replied:
Those local errors should be > 0, at least with the same likelihood as with other segments used elsewhere in the MuonIdProducer.

cvuosalo commented on TrackingTools/TrackAssociator/plugins/MuonDetIdAssociator.cc:
```diff
@@ -77,7 +79,14 @@ void MuonDetIdAssociator::getValidDetIds(unsigned int subDectorIndex, std::vecto
+
+  // ME0
+  if ( doME0_ ){
+    if (! geometry_->slaveGeometry(ME0DetId()) ) throw cms::Exception("FatalError") << "Cannnot ME0Geometry\n";
```
Exception message doesn't make sense and is misspelled. `Cannnot ME0Geometry` ???

cmsbuild commented:
-1 
 
Tested at: b6f5c54c8100eb44d6864b646b3d49c1d4b14610
I found errors in the following unit tests: 
 
 ---> test testRecoMETMETProducers had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13308/11254/summary.html

dnash86 commented:
@cvuosalo Regarding your first comment, about instructions for ME0-enabled-running:

The complete instructions are not quite complete yet, some "hacks" seem to be necessary to handle PhaseII in 80X.  We're working on this now, and decided to send the PR for the code in the meantime.  Can we provide them when they are complete?

Regarding your in-line comments: I will make your suggested changes and commit them.

Regarding the failed test:  It isn't 100% clear to me how the changes in this PR cause an exception in the module PFClusterProducer/'particleFlowClusterHF'.  Any suggestions?

cvuosalo commented:
@dnash86: The unit test error is an error in the IB, not this PR, so don't worry about it.

Please provide the instructions for enabling ME0 when they are ready. We need to verify that this PR works for ME0 muons.

dnash86 committed:
First iteration comments

cmsbuild commented:
Pull request #13308 was updated. @civanch, @cvuosalo, @mdhildreth, @cmsbuild, @slava77, @davidlange6 can you please check and sign again.

dnash86 commented:
I've pushed responses to most of your comments, other than a couple that I commented on in-line above.

dnash86 commented:
Let me know if I should go ahead and revert those inadvertent changes to the "fillmap" functionality, I am not 100% sure how that change was introduced, but I think it was an issue related to starting with an older version of the muonidproducer.

cvuosalo commented:
@dnash86: I don't understand your question regarding `MuonIdProducer.cc`. If you are saying that you put the wrong version of the file in this PR, why wouldn't you correct your mistake?

dnash86 commented:
@cvuosalo Agreed, odd question - I should clearly revert the change.  I simply did not realize that mistake was made until you commented on it, and was confused at first.  I'll work on reverting now.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13308/11254/summary.html

dnash86 committed:
Comments finished

cmsbuild commented:
Pull request #13308 was updated. @civanch, @cvuosalo, @mdhildreth, @cmsbuild, @slava77, @davidlange6 can you please check and sign again.

dnash86 commented:
@cvuosalo I've finished reverting the inadvertent changes to the MuonIdProducer.  The Muon.cc change was also performed, the "&&" should have been an "||".

cvuosalo commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: 1fee0a220afee4e8aa20a091ba1712804bac9676
I found errors in the following unit tests: 
 
 ---> test testRecoMETMETProducers had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13308/11279/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13308/11279/summary.html

cvuosalo commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -925,6 +969,64 @@ void MuonIdProducer::fillMuonId(edm::Event& iEvent, const edm::EventSetup& iSetu
+
+   		 //These matches are hardcoded, is that okay?
+   		 if( (AbsDx <= 3 or AbsDx/sqrt(localError.xx() + me0RecHit.localPositionError().xx()) <= 4) and 
+   		     (AbsDy <= 9999. or AbsDy/sqrt(localError.yy() + me0RecHit.localPositionError().yy()) <= 9999.)
```
Two questions about lines 1020-1021. Do you really want to allow `AbsDy > 9999`? It looks like 9999 is a max value, but the `or` in `AbsDy <= 9999. or AbsDy/sqrt(localError.yy() + me0RecHit.localPositionError().yy()) <= 9999.` would allow AbsDy to be larger.
Also, you can simply avoid any divide-by-0 by re-writing the checks. Instead of `AbsDx/sqrt(localError.xx() + me0RecHit.localPositionError().xx()) <= 4` just do:
`AbsDx <= 4.0 * sqrt(localError.xx() + me0RecHit.localPositionError().xx())`
No need to change the PR now unless you are making other changes.

dnash86 replied:
The point was to effectively not have a cut, but this is confusing.  The AbsDy line is removed in the upcoming push.

slava77 commented:
- The PR subject is misleading. The code is actually implementing tracker muons (not global muons == combined fit of standalone and inner tracker track)
- is it clear why the ME0 added in parallel? It's a detector that delivers segments in one (and only) station. we could treat this as station 5 to be independent of everything else (so that the region of ~2.0 to 2.4 in eta is not ambiguous with station 1 of CSC). There doesn't seem to be a c++ issue to keep ME0 matches in the same collection. It may be better to keep them in segmentMatches. The code will be simpler.

slava77 commented on DataFormats/MuonReco/interface/Muon.h:
```diff
@@ -175,15 +175,17 @@ namespace reco {
 
     /// define arbitration schemes
-    enum ArbitrationType { NoArbitration, SegmentArbitration, SegmentAndTrackArbitration, SegmentAndTrackArbitrationCleaned, RPCHitAndTrackArbitration };
+    enum ArbitrationType { NoArbitration, SegmentArbitration, SegmentAndTrackArbitration, SegmentAndTrackArbitrationCleaned, RPCHitAndTrackArbitration, ME0HitAndTrackArbitration };
```
Why "Hit and Track"
 if I'm not mistaken, the objects used in matching are segments

dnash86 replied:
Agreed, this is changed to "SegmentAndTrack" in the upcoming push.

slava77 commented on DataFormats/MuonReco/src/Muon.cc:
```diff
@@ -147,6 +166,23 @@ unsigned int Muon::stationMask( ArbitrationType type ) const
+      	      me0Match != chamberMatch->me0Matches.end(); me0Match++ )
+      	   {
+      	     //Not sure if mask is correct..
+      	     curMask = 1<<( (chamberMatch->station()-1)+4*(chamberMatch->detector()-1) );
```
There is only one station (1), and the detector is always 5, if it's actually ME0.
So, the result is (0 + 16). curMask should correspond to a sensible detector layer that can be assigned to the place where the segments are.
In the main thread I suggested 5.

slava77 replied:
... actually, 5 is not that good. 
If you keep me0Matches separately, then this can be simply 1; if the me0 segments are merged with regular (DT/CSC) segment matches, then this can be 8 (DTs go from 0 to 3; CSC from 4 to 7).

slava77 commented on DataFormats/MuonReco/src/Muon.cc:
```diff
@@ -213,6 +249,21 @@ int Muon::numberOfMatchedRPCLayers( ArbitrationType type ) const
+{
+   int layers(0);
+
+   unsigned int theME0LayerMask = ME0layerMask(type);
```
This doesn't make much sense considering that ME0 segments are reconstructed in the full ME0 (single station).
Layers are needed if the match ID is to be computed in individual layers (as RPCs are matched separately)

slava77 commented on DataFormats/MuonReco/src/classes_def.xml:
```diff
@@ -42,11 +42,16 @@
    <version ClassVersion="10" checksum="1493325087"/>
   </class>
-  <class name="reco::MuonChamberMatch" ClassVersion="11">
+  <class name="reco::MuonChamberMatch" ClassVersion="15">
```
please do not introduce intermediate versions that you had while developing the code.
Since 11 was the last one, drop the current 12,13,14 and rename the current 15 to be 12.

slava77 commented on RecoMuon/DetLayers/src/MuonME0DetLayerGeometryBuilder.cc:
```diff
@@ -73,7 +73,7 @@ MuonME0DetLayerGeometryBuilder::buildLayer(int endcap,
       
     for(std::vector<int>::iterator chamber=chambers.begin(); chamber<chambers.end(); chamber++) {
-      ME0DetId me0Id(endcap,1,(*chamber), 0);
+      ME0DetId me0Id(endcap,1,(*chamber), 1);  
```
Please clarify why this was needed.
Wouldn't this correspond to the roll/partition 1 volume, which has smaller boundaries than the full chamber. (reco occasionally uses the boundaries of detector geometry elements)
I'm not sure if this makes any difference in this code though.

slava77 commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -685,9 +712,10 @@ void MuonIdProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
      reco::MuonTimeExtra combinedTime;
 
-     theTimingFiller_->fillTiming(muon, dtTime, cscTime, combinedTime, iEvent, iSetup);
+     theTimingFiller_->fillTiming(muon, dtTime, cscTime, rpcTime, combinedTime, iEvent, iSetup);
```
rpcTime should be already in the baseline.
Looks like an old version of the code is seen by at least the github display.

dnash86 replied:
Yes, I saw it in the baseline as well.  I am confused, is there something wrong here?

slava77 replied:
looking at your commit history there is a back-and-forth with the old and new version of these lines.
a rebase (to 800 or at least pre6) may help (if you manage to get past the merge commit in the middle of your branch .. there is just one of them, which likely means the rebase will work).

slava77 commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -750,6 +779,16 @@ bool MuonIdProducer::isGoodRPCMuon( const reco::Muon& muon )
+bool MuonIdProducer::isGoodME0Muon( const reco::Muon& muon )
+{
+  if(muon.track()->pt() < minPt_ || muon.track()->p() < minP_) return false;
+  //These "ExtraSoftMuons" may not be relevant, since there's no eta coverage below eta 1.5 anyway - the eta requirement should always fail
```
better drop this "if" in this case. (it's always false due to the eta range)

dnash86 replied:
This if is dropped for now, the cuts may need to be re-examined later.

slava77 commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -750,6 +779,16 @@ bool MuonIdProducer::isGoodRPCMuon( const reco::Muon& muon )
 
+bool MuonIdProducer::isGoodME0Muon( const reco::Muon& muon )
+{
+  if(muon.track()->pt() < minPt_ || muon.track()->p() < minP_) return false;
```
minP_ and minPt_ were derived to match the lowest penetrating momentum in eta up to 2.4. 
Please check that it still applies for ME0.
The version going to eta ~4 (if it ever becomes real) will end up being cut too much.
A different set of cuts may be needed.

slava77 commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -750,6 +779,16 @@ bool MuonIdProducer::isGoodRPCMuon( const reco::Muon& muon )
+   if ( addExtraSoftMuons_ && 
+	muon.pt()<5 && fabs(muon.eta())<1.5 && 
+	muon.numberOfMatchedME0Layers( reco::Muon::ME0HitAndTrackArbitration ) > 0 ) return true;
+   return ( muon.numberOfMatchedME0Layers( reco::Muon::ME0HitAndTrackArbitration ) >= minNumberOfMatches_ );
```
this selection is not practical considering that ME0 is actually just one station

slava77 commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.h:
```diff
@@ -204,15 +211,18 @@ class MuonIdProducer : public edm::stream::EDProducer<> {
    edm::EDGetTokenT<edm::ValueMap<reco::MuonQuality> > glbQualToken_;
 
    edm::Handle<RPCRecHitCollection> rpcHitHandle_;
+   edm::Handle<ME0SegmentCollection> me0HitHandle_;
```
it's confusing to call the segment collection  a "hit handle"

slava77 commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -808,6 +848,10 @@ void MuonIdProducer::fillMuonId(edm::Event& iEvent, const edm::EventSetup& iSetu
      if (chamber.id.subdetId() == 3 && rpcHitHandle_.isValid()  ) continue; // Skip RPC chambers, they are taken care of below)
+
+     if (doME0_){
+       if  (chamber.id.subdetId() == MuonSubdetId::ME0 && me0HitHandle_.isValid()  ) continue; // Skip ME0 chambers, they are taken care of below
```
shouldn't this skip happen regardless of doME0_ value?
Otherwise, if doME0_ is false, you will include ME0 chambers here

dnash86 replied:
If the doME0_ is disabled, there won't be any ME0s at all.  If it is enabled, it should still skip non-ME0 chambers.  Should the doME0_ be removed in any case?  My intention was to not check the handle if it wouldn't be there.

slava77 commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -1234,3 +1336,13 @@ bool MuonIdProducer::checkLinks(const reco::MuonTrackLinks* links) const {
 }
+
+/// ParameterSet descriptions
+void MuonIdProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
```
this better include all parameters for this module.
... maybe OK.

slava77 commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.h:
```diff
@@ -204,15 +211,18 @@ class MuonIdProducer : public edm::stream::EDProducer<> {
    reco::isodeposit::IsoDepositExtractor* muIsoExtractorCalo_;
    reco::isodeposit::IsoDepositExtractor* muIsoExtractorTrack_;
    reco::isodeposit::IsoDepositExtractor* muIsoExtractorJet_;
+   //bool doME0_;
```
remove this here?

slava77 commented on SLHCUpgradeSimulations/Configuration/python/fixMissingUpgradeGTPayloads.py:
```diff
@@ -19,3 +19,33 @@ def fixRPCConditions(process):
     return process
 
 
+def fixDTAlignmentConditions(process):
```
are these functions needed?
Which GT are you using for testing?

Neither of these functions are used in this PR.

dnash86 committed:
second round comments, remove me0Matches and LayerMask

cmsbuild commented:
Pull request #13308 was updated. @civanch, @cvuosalo, @mdhildreth, @cmsbuild, @slava77, @davidlange6 can you please check and sign again.

dnash86 commented:
@slava77 Sorry about the misleading title, you are correct.  As for your other general comment, we took your advice and removed the "me0Matches" variable, now simply storing matches to ME0 as segmentMatches.  The proper arbritration is handled by the ME0SegmentAndTrackArbitration type, using the subdetId of the matched chamber to check that it is an ME0.

slava77 commented on RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:
```diff
@@ -808,6 +849,10 @@ void MuonIdProducer::fillMuonId(edm::Event& iEvent, const edm::EventSetup& iSetu
    {
      if (chamber.id.subdetId() == 3 && rpcHitHandle_.isValid()  ) continue; // Skip RPC chambers, they are taken care of below)
+
+     if (doME0_){
```
a separate parameter in a separate place controls what goes into info.chambers.
It's better to not assume everything is set correctly and just skip the me0segments here completely.
Currently if the MuonIdProducer doMe0 is false, but if the associator doME0 is true, you will end up with ME0 chambers parsed in this block (and also in the specific ME0 block as well).

dnash86 replied:
Ah, that is a good point.  Should I remove the "if (doME0_)" part here?  Which other specific ME0 block did you mean?

slava77 replied:
Yes, please remove the ``` if (doME0_) ```. 

By the specific ME0 block, I meant the code that follows after ``` // Fill ME0 info ``` on line 974 and below.

dnash86 replied:
@slava77 Sorry for the delay, done now.

slava77 commented:
On 2/18/16 9:28 AM, dnash86 wrote:
> @slava77 <https://github.com/slava77> Sorry about the misleading title,
> you are correct. 

David,
please edit the PR subject (there is an "Edit" button at the top  right
of the web page).


Thanks.

> As for your other general comment, we took your advice
> and removed the "me0Matches" variable, now simply storing matches to ME0
> as segmentMatches. The proper arbritration is handled by the
> ME0SegmentAndTrackArbitration type, using the subdetId of the matched
> chamber to check that it is an ME0.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/13308#issuecomment-185825665>.
>

dnash86 commented:
@slava77 I changed the PR subject, is this one okay?  Sorry, the branch still has the misleading name...  

I also added a comment in response to your inline comment, above.

slava77 commented:
branch topic names are somewhat free-form.
It's important to have a good descriptive title/subject for  a PR, since it's used for release notes.

cvuosalo commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: 9eb7733d5a34db58a0865ae0772138726f8828a5
I found errors in the following unit tests: 
 
 ---> test runtestTqafTopEventSelection had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13308/11353/summary.html

dnash86 committed:
third comment, remove doME0_ from a couple points

cmsbuild commented:
Pull request #13308 was updated. @civanch, @cvuosalo, @mdhildreth, @cmsbuild, @slava77, @davidlange6 can you please check and sign again.

dnash86 commented:
Performed the above changes regarding doME0_ in the MuonIdProducer requested by @slava77.  Sorry for the delay.

cvuosalo commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: 31e53d02216d28158d787d7dae403a0076f4dbbd
I found errors in the following unit tests: 
 
 ---> test runtestTqafTopHitFit had ERRORS
---> test runtestTqafTopEventSelection had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13308/11393/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13308/11393/summary.html

cvuosalo commented on DataFormats/MuonReco/interface/MuonChamberMatch.h:
```diff
@@ -12,6 +12,9 @@ namespace reco {
          std::vector<reco::MuonSegmentMatch> truthMatches;      // SimHit projection matching propagated track trajectory
          std::vector<reco::MuonRPCHitMatch>  rpcMatches;        // rpc hits matching propagated track trajectory 
+	 //double testing;
+	 //double testingtoo;
```
Please delete commented-out lines of code.

cvuosalo commented:
@dnash86: Please send complete instructions for testing with ME0 muons enabled. That is the only issue holding up this PR, other than the minor comment clean-up noted above. Jenkins unit test errors are in the IB, not this PR. Jenkins results are OK, and the code changes are OK now.

cvuosalo commented:
@dnash86: Ping. When can you provide instructions for testing?

cvuosalo commented:
@dnash86: Ping. Please let us know your plans for this PR. Can you provide testing instructions?

davidlange6 commented:
this PR is not needed in 80x (cleaning up the 80x queue)
